bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -330,24 +330,12 @@ private boolean replaceSelection(String replaceString) {
 	}
 
 	@Override
-	public boolean performSearch(String searchString) {
-		return performSearch(shouldInitIncrementalBaseLocation(), searchString);
-	}
-
-	/**
-	 * Locates the user's findString in the text of the target.
-	 *
-	 * @param mustInitIncrementalBaseLocation <code>true</code> if base location
-	 *                                        must be initialized
-	 * @param findString                      the String to search for
-	 * @return Whether the string was found in the target
-	 */
-	private boolean performSearch(boolean mustInitIncrementalBaseLocation, String findString) {
-		if (mustInitIncrementalBaseLocation) {
-			resetIncrementalBaseLocation();
-		}
+	public boolean performSearch(String findString) {
 		resetStatus();
 
+		if (isActive(SearchOptions.INCREMENTAL) && !isIncrementalSearchAvailable()) {
```

Comment:
To me, it looks as if this should do basically the same than `shouldInitIncrementalBaseLocation()`, only that the other method has a simpler check for regex option being activated. I think this implementation is "more correct", so maybe they should/could be combined to one correct version?

Reply:
No, this isn't quite the same. The check here checks:

isActive(Incremental) AND is unavailable

the check in shouldInitIncrementalBaseLocation checks for

isActive(Incremental) AND is available

regardless, shouldInitIncrementalBaseLocation changes anyway in this PR: https://github.com/eclipse-platform/eclipse.platform.ui/pull/2108. With that change, I don't think your comment still applies

Reply:
I see, thank you for the explanation.